bitnami/zipkin/templates/_helpers.tpl

```diff
@@ -0,0 +1,254 @@
+{{/*
+Copyright Broadcom, Inc. All Rights Reserved.
+SPDX-License-Identifier: APACHE-2.0
+*/}}
+
+{{/*
+Return the proper zipkin image name
+*/}}
+{{- define "zipkin.image" -}}
+{{ include "common.images.image" (dict "imageRoot" .Values.image "global" .Values.global) }}
+{{- end -}}
+
+{{/*
+Return the proper image name (for the wait init container)
+*/}}
+{{- define "zipkin.init-containers.wait.image" -}}
+{{ include "common.images.image" (dict "imageRoot" .Values.defaultInitContainers.wait.image "global" .Values.global) }}
+{{- end -}}
+
+{{/*
+Return the proper Docker Image Registry Secret Names
+*/}}
+{{- define "zipkin.imagePullSecrets" -}}
+{{- include "common.images.renderPullSecrets" (dict "images" (list .Values.image .Values.defaultInitContainers.wait.image) "context" $) -}}
+{{- end -}}
+
+{{/*
+Create the cassandra host
+*/}}
+{{- define "zipkin.cassandra.host" -}}
+    {{- if not .Values.cassandra.enabled -}}
+        {{- .Values.externalDatabase.host | quote -}}
+    {{- else -}}
+        {{- include "common.names.dependency.fullname" (dict "chartName" "cassandra" "chartValues" .Values.cassandra "context" $) -}}
+    {{- end }}
+{{- end }}
+
+{{/*
+Return the cassandra Port
+*/}}
+{{- define "zipkin.cassandra.port" -}}
+{{- if .Values.cassandra.enabled }}
+    {{- print .Values.cassandra.service.ports.cql -}}
+{{- else -}}
+    {{- printf "%d" (.Values.externalDatabase.port | int ) -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return the cassandra datacenter
+*/}}
+{{- define "zipkin.cassandra.datacenter" -}}
+{{- if .Values.cassandra.enabled }}
+    {{- print .Values.cassandra.cluster.datacenter -}}
+{{- else -}}
+    {{- print .Values.externalDatabase.cluster.datacenter -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return the cassandra Database Name
+*/}}
+{{- define "zipkin.cassandra.keyspace" -}}
+{{- if .Values.keyspace }}
+    {{- /* Inside cassandra subchart */ -}}
+    {{- print .Values.keyspace -}}
```

Comment:
The value `.Values.keyspace` does not exist in the values.yaml.

Could you please provide more information about what you mean with 'Inside cassandra subchart'?

Reply:
It's because of this:

```
cassandra:
  initDB:
    create_keyspace.cql: |
      CREATE KEYSPACE {{ include "zipkin.cassandra.keyspace" . }} WITH REPLICATION = { 'class' : 'SimpleStrategy', 'replication_factor' : 1 };
```

The helper `zipkin.cassandra.keyspace` is executed inside the `cassandra` scope, so `.Values.cassandra.keyspace` does not exist. Instead, the existing value is `.Values.keyspace`

Reply:
Understood, thanks!